src/App.tsx

```diff
@@ -15,37 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  SORT_BY_CHAR = 'SORT_BY_CHAR',
+  SORT_BY_LENGTH = 'SORT_BY_LENGTH',
+}
+
+type TGoods = string[];
+
+interface IGoodsProps {
+  goods: Readonly<TGoods>;
+}
+
+const Goods: FC<IGoodsProps> = ({ goods }) => (
+  <ul>
+    {goods.map(el => (
+      <li data-cy="Good" key={el}>
+        {el}
+      </li>
+    ))}
+  </ul>
+);
+
+type TSortGoods = (
+  goods: TGoods,
+  sortType: typeof SortType | string,
+  isReverse: boolean,
+) => TGoods;
+
+const sortGoods: TSortGoods = (goods, sortType, isReverse = false) => {
+  if (!sortType && isReverse) return [...goods].reverse();
+
+  const sortedGoods = [...goods].sort((a, b) => {
+    switch (sortType) {
+      case SortType.SORT_BY_CHAR:
+        return isReverse ? b.localeCompare(a) : a.localeCompare(b);
+
+      case SortType.SORT_BY_LENGTH:
+        return a.length - b.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReverse && sortType === SortType.SORT_BY_LENGTH
+    ? sortedGoods.reverse()
+    : sortedGoods;
+};
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReverse, setReverse] = useState(false);
+
+  const goods = sortGoods(goodsFromServer, sortType, isReverse);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortType !== SortType.SORT_BY_CHAR,
+          })}
+          onClick={() => setSortType(SortType.SORT_BY_CHAR)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortType !== SortType.SORT_BY_LENGTH,
+          })}
+          onClick={() => setSortType(SortType.SORT_BY_LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReverse,
+          })}
+          onClick={() => {
+            setReverse(!isReverse);
+          }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortType !== '' || isReverse) && (
```

Comment:
will be the same, but syntax is better
```suggestion
        {(sortType.length || isReverse) && (
```